wiki/Oppia-Android-Code-Coverage.md

```diff
@@ -0,0 +1,380 @@
+```
+
+By default, this will generate an HTML report in the coverage_reports directory. For the given file, the report will be saved as **coverage_reports/utility/src/main/java/org/oppia/android/util/math/MathTokenizer/coverage.html**
+
+A list of files can be provided as an input to generate coverage reports for all the files.
+
+```
+bazel run //scripts:run_coverage -- $(pwd) 
+utility/src/main/java/org/oppia/android/util/math/MathTokenizer.kt 
+utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt
+```
+
+### Process Timeout
+
+The default wait time for a process to complete is 300 seconds (5 minutes). If the process does not finish within this period, you may encounter a "Process Did Not Finish Within Intended Time" error. To extend the wait time, you can modify the timeout setting by adding the flag --processTimeout=15 (The Time unit is minutes).
+
+```
+bazel run //scripts:run_coverage -- $(pwd)
+utility/src/main/java/org/oppia/android/util/parser/math/MathModel.kt --processTimeout=15
+```
+
+## 2.1 Understanding the CI Coverage Report
+
+Let's examine the coverage.html report located at coverage_reports/utility/src/main/java/org/oppia/android/util/math/MathTokenizer/coverage.html to understand its contents.
+
+**Header Section:** At the top of the report, you'll find information including:
+
+![image](https://github.com/user-attachments/assets/08c4f0a0-51b5-4d8e-a2f3-ff13d8b6cb2f)
+
+#
+
+- **File Path:** The relative path of the file to the oppia-android codebase.
+- **Coverage Percentage:** The percentage of code covered by tests.
+- **Line Coverage Data:** Details on the coverage of individual lines.
+- **Main Content:**
+
+Visual Coverage Representation: The report visually highlights which lines are covered and which are not.
+
+![image](https://github.com/user-attachments/assets/0e36fde3-639b-4874-b809-59d33827388d)
+
+
+#
+
+- **Red Lines:** Indicate lines not covered by test cases.
+- **Green Lines:** Represent lines that are covered by test cases.
+
+These generated html reports are be used to identify areas of your code that may need additional testing.
+
+## Increasing Code Coverage Metrics
+
+While figuring out which lines are covered and which are not, you can increase code coverage by ensuring that each line is thoroughly tested. To enhance code coverage effectiveness, locate the test files associated with the source file (typically found in the same package but within the test directories).
+
+Note: Some files in the app module may have multiple test files, located in the sharedTest and test packages, all testing a single source file. (eg: The file StateFragment.kt has 2 test files StateFragmentTest.kt under sharedTest and StateFragmentLocalTest.kt under test folder)
+
+By identifying and adding appropriate test scenarios to cover previously uncovered lines, you will boost the coverage percentage and improve the overall quality and reliability of your code.
```

Comment:
This section on approval will add a link to the writing effective tests or writing tests with good behavioural coverage page.

Reply:
I think two paragraphs above might be a good place for the link (see my comment). Maybe also move this sentence to that paragraph or drop it since the section will probably be a bit easier to read with the 'note' being the last item.

Reply:
Done, rephrased the above paragraph to include this too thereby positioning the `Note` section to the end.

Reply:
Great--do you think this can be resolved now or is further action needed?

Reply:
Thanks, resolving it with the link added to the referenced wiki - https://github.com/oppia/oppia-android/pull/5483/files#diff-696bbeaab338ccd0f2816eba3863990fe6fad20ec44ef5bec8475950eea26b76R377